driver-core/src/test/unit/com/mongodb/internal/connection/ByteBufferBsonInputTest.java

```diff
@@ -54,37 +54,68 @@ class ByteBufferBsonInputTest {
-                size -> new ByteBufNIO(ByteBuffer.allocateDirect(size)),
-                size -> new ByteBufNIO(ByteBuffer.allocate(size)) {
-                    @Override
-                    public boolean isBackedByArray() {
-                        return false;
-                    }
-
-                    @Override
-                    public byte[] array() {
-                        return Assertions.fail("array() is called, when isBackedByArray() returns false");
-                    }
-
-                    @Override
-                    public int arrayOffset() {
-                        return Assertions.fail("arrayOffset() is called, when isBackedByArray() returns false");
-                    }
-                }
+                createBufferProvider(
+                        "NettyByteBuf based on PooledByteBufAllocator.DEFAULT.directBuffer",
+                        size -> new NettyByteBuf(PooledByteBufAllocator.DEFAULT.directBuffer(size))
+                ),
+                createBufferProvider(
+                        "NettyByteBuf based on PooledByteBufAllocator.DEFAULT.heapBuffer",
+                        size -> new NettyByteBuf(PooledByteBufAllocator.DEFAULT.heapBuffer(size))
+                ),
+                createBufferProvider(
+                        "PowerOfTwoBufferPool",
+                        new PowerOfTwoBufferPool()
+                ),
+                createBufferProvider(
+                        "ByteBufNIO based on ByteBuffer with arrayOffset() -> 2",
+                        size -> new ByteBufNIO(ByteBuffer.wrap(new byte[size + 5], 2, size).slice())
+                ),
+                createBufferProvider(
+                        "ByteBufNIO based on ByteBuffer with arrayOffset() -> 3,",
+                        size -> new ByteBufNIO(ByteBuffer.wrap(new byte[size + 4], 3, size).slice())
+                ),
+                createBufferProvider(
+                        "ByteBufNIO emulating direct ByteBuffer",
+                        size -> new ByteBufNIO(ByteBuffer.allocate(size)) {
+                            @Override
+                            public boolean isBackedByArray() {
+                                return false;
+                            }
+
+                            @Override
+                            public byte[] array() {
+                                return Assertions.fail("array() is called, when isBackedByArray() returns false");
+                            }
+
+                            @Override
+                            public int arrayOffset() {
+                                return Assertions.fail("arrayOffset() is called, when isBackedByArray() returns false");
+                            }
+                        }
+                )
         );
     }
 
-    @ParameterizedTest
+    private static BufferProvider createBufferProvider(final String bufferName, final BufferProvider bufferProvider) {
```

Comment:
I am not sure about my point. Is it true that bufferDescription is more appropriate naming? Saw it is used to overwrite toString.